src/pubkey.cpp

```diff
@@ -65,14 +66,15 @@ static int ecdsa_signature_parse_der_lax(const secp256k1_context* ctx, secp256k1
     lenbyte = input[pos++];
     if (lenbyte & 0x80) {
         lenbyte -= 0x80;
-        if (pos + lenbyte > inputlen) {
+        if (lenbyte > inputlen - pos) {
             return 0;
         }
         while (lenbyte > 0 && input[pos] == 0) {
             pos++;
             lenbyte--;
         }
-        if (lenbyte >= sizeof(size_t)) {
+        static_assert(sizeof(size_t) >= 4, "size_t too small");
+        if (lenbyte >= 4) {
             return 0;
```

Comment:
If ``lenbyte >= 4`` here, then ``rlen`` calculated in the ``while`` loop just below will be >= 2<sup>32</sup> (because ``lenbyte`` at this point is the length of the representation of ``rlen`` after stripping leading zeros). So this function would in any case return 0 at line 90 if it does not return 0 here: the condition ``rlen > inputlen - pos`` would be true at line 90 since ``inputlen`` is at most the maximum block size, which is much less than 2^32 bytes.

Therefore, there is no consensus change (it doesn't matter whether we return 0 here or at line 90).

If this had actually made a difference, then there would have been a consensus incompatibility between 32-bit and 64-bit platforms, which would surely have been unintended. (I know that originally only 64-bit platforms were supported, but my understanding is that the current code is supposed to support both with no consensus differences.)